pallets/capacity/src/lib.rs

```diff
@@ -1024,37 +1053,42 @@ impl<T: Config> Pallet<T> {
 			Some(provider_boost_history) => {
 				match provider_boost_history.count() {
 					0usize => false,
-					// they staked before the current era, so they have unclaimed rewards.
-					1usize => provider_boost_history.get_entry_for_era(&current_era).is_none(),
+					1usize => {
+						// if there is just one era entry and:
+						// it's for the previous era, it means we've already paid out rewards for that era, or they just staked in the last era.
+						// of if it's for the current era, they only just started staking.
+						provider_boost_history
+							.get_entry_for_era(&current_era.saturating_sub(1u32.into()))
+							.is_none() && provider_boost_history
+							.get_entry_for_era(&current_era)
+							.is_none()
```

Comment:
I might got confused but what about the case that there are 2 entries and 1 of them in `current_era` and the other one in `current_era -1 ` ? What should happen in that case?

Reply:
current era is not paid out as rewards, because you only get rewards for amounts you stake in an entire era.